File: app/src/main/java/com/tstudioz/iksica/Fragments/ProfileFragment.java
package com.tstudioz.iksica.Fragments;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.mikhaellopez.circularimageview.CircularImageView;
import com.tstudioz.iksica.Activities.SignInActivity;
import com.tstudioz.iksica.Adapter.AdapterInfo;
import com.tstudioz.iksica.Model.User;
import com.tstudioz.iksica.Model.UserInfoItem;
import com.tstudioz.iksica.R;

import io.realm.Realm;
import io.realm.RealmResults;

/**
 * Created by etino7 on 18-Oct-17.
 */

public class ProfileFragment extends Fragment {

    private RecyclerView rv;
    private Button signOutButton;
    Realm mRealm;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedBundleInstance){
        View view = inflater.inflate(R.layout.profile_layout, parent, false);

        mRealm = Realm.getDefaultInstance();
        User user = mRealm.where(User.class).findFirst();

        CircularImageView image = (CircularImageView) view.findViewById(R.id.circularImageView);
        Glide.with(this).load(user.getSrcLink()).into(image);

        TextView imeStudenta = (TextView)view.findViewById(R.id.name_surename);
        imeStudenta.setText(user.getuName());

        rv = (RecyclerView)view.findViewById(R.id.user_info_recycler);
        showInfoRecyclerView();

        signOutButton = (Button)view.findViewById(R.id.button_logout);
        signOutButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                signOut();
            }
        });

        return view;
    }

    public void showInfoRecyclerView(){
        RealmResults<UserInfoItem> informacije = mRealm.where(UserInfoItem.class).findAll();

        rv.setHasFixedSize(true);
        rv.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, false));
        AdapterInfo ah = new AdapterInfo(informacije);
        rv.setAdapter(ah);
    }

    public void signOut(){
        SharedPreferences sp = getActivity().getSharedPreferences("SHARED_PREFS", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sp.edit();
        editor.remove("korisnik_prijavljen");
        editor.commit();

        mRealm.executeTransaction(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                mRealm.deleteAll();
            }
        });

        getActivity().startActivity(new Intent(getActivity(), SignInActivity.class));
        getActivity().finish();
    }

    @Override
    public void onDestroy(){
        super.onDestroy();

        if(mRealm!=null)
            mRealm.close();
    }
}

File: app/src/main/java/com/tstudioz/iksica/Fragments/ProfileFragment.kt
package com.tstudioz.iksica.Fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import com.bumptech.glide.Glide
import com.tstudioz.iksica.Adapter.AdapterInfo
import com.tstudioz.iksica.CardScreen.MainViewModel
import com.tstudioz.iksica.Data.Models.PaperUser
import com.tstudioz.iksica.Data.Models.UserInfoItem
import com.tstudioz.iksica.R
import kotlinx.android.synthetic.main.profile_layout.*
import timber.log.Timber

/**
 * Created by etino7 on 18-Oct-17.
 */

class ProfileFragment : Fragment() {

    var viewModel: MainViewModel? = null

    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, savedBundleInstance: Bundle?): View? {
        val view = inflater.inflate(R.layout.profile_layout, parent, false)

        viewModel = ViewModelProvider(activity!!)[MainViewModel::class.java]

        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        recyclerProfile.layoutManager = GridLayoutManager(view.context, 2, GridLayoutManager.VERTICAL, false)


        viewModel?.getUserData()?.observe(viewLifecycleOwner, Observer {
            it?.let {
                Glide.with(view.context)
                        .load(it.avatarLink)
                        .into(avatar)
                name_surname.text = it.name
                user_desc.text = getString(R.string.student_desc).plus(" ").plus(it.university)

                val recyclerViewdata: ArrayList<UserInfoItem> = convertUserDataToList(it)
                Timber.d(recyclerViewdata[0].item)
                val adapter = AdapterInfo(recyclerViewdata)
                recyclerProfile.adapter = adapter
            }
        })
    }

    private fun convertUserDataToList(user: PaperUser): ArrayList<UserInfoItem> {

        val razinaPrava = UserInfoItem("Razina prava", user.rightsLevel.toString())
        val pravaOd = UserInfoItem("Prava od", user.rightsFrom)
        val pravaDo = UserInfoItem("Prava do", user.rightsTo)

        return arrayListOf<UserInfoItem>(razinaPrava, pravaOd, pravaDo)
    }


}
